### PR TITLE
Centralize conf

### PR DESCRIPTION
**Pull request by gturri**
Goals:
- There's only one conf file to change
- It seems we forgot the port in the html files. I added it. (correct me if I was wrong)

Next steps would be:
- Give the ability to have a `config.local.js` which could override the default `config.js` and be gitignored
- Let "http://" be in the config file (I've let it outside because it seems to mess things up because of slashes which are replaced by their ascii code. (probable just a mustache flag, I just didn't had the time to look into it yet)

### CODE REVIEW

**Comment by AlexFaraino** [comment]
As this is a tiny application, many will use it behind a reverse proxy. Meaning, the port node listens on won't be the same the client will connect to.
i.e. nodejs listens on 8081 and a reverse proxy (Apache + mod_proxy, HA_proxy, Varnish, ...) listens on 80 and forwards requests for "door.mydomain.tld" to localhost:8081.
In this case, we can add two params in config file:
- `reverse-proxy-enable` (= 0 or `false` by default)
- `reverse-proxy-port` (=80 by default)

Great Job BTW. Thx

**Comment by AlexFaraino** [comment]
And about the config files, we can just set the defaults in config.js.template. If config.js doesn't exists, nodejs can copy the template to config.js and tell users to make changes in this file.

File: index.js [hunk 1]
```diff
@@ -1,19 +1,32 @@
 // Init Express Framework
 var express = require("express");
+var fs = require("fs");
+var mustache = require("mustache");
+var config = require("./config");
 var app = express();
-var port = 8081;
+var port = config.port;
+var url = config.host + ":" + port;
```
Comment (AlexFaraino) [inline index.js]: Here, the url can be:
`var url = config.host + ":" + (config.reverse-proxy ? config.reverse-proxy-port : port);`

File: config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+var config = {}
+
+config.port = 8081;
+config.host = "door.fabelier.org"
+
```
Comment (AlexFaraino) [inline config.js]: To support reverse proxy:
- `config.reverse-proxy = 0;`
- `config.reverse-proxy-port = 80;`

**Comment by gturri** [comment]
Thanks for the explanation, I was unaware of the reverse proxy subtlety. Here is an updated of the PR with your suggestions.

**Pull request merged**